kolibri/core/auth/management/commands/bulkimportusers.py

```diff
@@ -0,0 +1,587 @@
+import csv
+import logging
+import re
+import sys
+
+from django.contrib.auth.hashers import make_password
+from django.core.exceptions import ValidationError
+from django.core.management.base import CommandError
+
+from kolibri.core.auth.constants.demographics import choices
+from kolibri.core.auth.csv_utils import input_fields
+from kolibri.core.auth.csv_utils import labels
+from kolibri.core.auth.models import Classroom
+from kolibri.core.auth.models import Facility
+from kolibri.core.auth.models import FacilityUser
+from kolibri.core.auth.models import Membership
+from kolibri.core.tasks.management.commands.base import AsyncCommand
+from kolibri.core.tasks.utils import get_current_job
+
+logger = logging.getLogger(__name__)
+DEFAULT_PASSWORD = make_password("kolibri")
+fieldnames = (
+    "Username",
+    "Password",
+    "Full name",
+    "User type",
+    "Identifier",
+    "Birth year",
+    "Gender",
+    "Enrolled in",
+    "Assigned to",
+)
+roles_map = {
+    "learner": None,
+    "admin": "admin",
+    "facility coach": "coach",
+    "class coach": "classroom assignable coach",
+}
+
+# Validators ###
+
+
+def number_range(min, max):
+    """
+    Return a value check function which raises a ValueError if the supplied
+    value is less than `min` or greater than `max`.
+    """
+
+    def checker(v):
+        if int(v) < min or int(v) > max:
+            raise ValueError(v)
+
+    return checker
+
+
+def value_length(length, null=False):
```

Comment:
I found the expression `if null` a little confusing.

Maybe

```suggestion
def value_length(length, allow_null=False):
```

which would make it:

```python
if allow_null and v is None:
```

Reply:
thanks, makes sense